ffi/capi/dart/package/test/test.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:icu/lib.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+void main() {
+  init(path.absolute("test/libicu_capi_cdylib.so"));
+
+  test('ICU4XFixedDecimal.toString', () {
+    final x = ICU4XFixedDecimal.fromF64WithLowerMagnitude(1.49403, -7);
+    expect(x.toString(), "1.4940300");
+    print(x.toString());
+  });
+
+  test('ICU4XLocaleFallbacker', () {
+    final iterator = ICU4XLocaleFallbacker(ICU4XDataProvider.compiled())
+        .forConfig(ICU4XLocaleFallbackConfig()
+          ..extensionKey = "ca"
+          ..priority = ICU4XLocaleFallbackPriority.Region)
+        .fallbackForLocale(ICU4XLocale.fromString("de-CH-u-ca-japanese"));
+    while (iterator.get.toString() != "und") {
```

Comment:
This fails when replacing this line with `while (iterator.get != ICU4XLocale.und()) {` -- why?

Also, the test doesn't contain an `expect`.

Reply:
Ah, two things at play here: 
* I forgot to enable release mode for the Rust library, and that enables some debug assertions. With release mode, this becomes an infinite loop because
* I cannot overwrite `==` so `iterator.get != ICU4XLocale.und()` will always be true

Reply:
Could diplomat generate something like 

```dart
  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;

    return other is ICU4XLocale && other.toString() == toString();
  }

  @override
  int get hashCode => _underlying.hashCode;
```
in `ICU4XLocale`?

Reply:
We actually have a semantic equality method in Rust, but that's not currently exposed through Diplomat. We need to teach Diplomat about special equality functions, then we can generate `==` for opaque structs. Going through `toString` is not a good idea because it's not very efficient.

Reply:
Even better!